File: src/backend_dotnet/HRMApi/Models/Activity.cs
using System;
using System.Collections.Generic;

namespace HRMApi.Models;

public partial class Activity
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public DateTime RegistrationStartDate { get; set; }

    public DateTime RegistrationEndDate { get; set; }

    public int? MaxParticipants { get; set; }

    public string? Location { get; set; }

    public string? ActivityType { get; set; }

    public string? ImageUrl { get; set; }

    public string? Organizer { get; set; }

    public int? Points { get; set; }

    public string? Status { get; set; }

    public int? CreatedBy { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public bool? IsDeleted { get; set; }

    public virtual Employee? CreatedByNavigation { get; set; }

    public virtual ICollection<Participation> Participations { get; set; } = new List<Participation>();
}

File: src/backend_java/src/main/java/com/group07/human_resource_management/entity/Activity.java
package com.group07.human_resource_management.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "activity")
@Getter @Setter
@Builder @NoArgsConstructor @AllArgsConstructor
public class Activity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String description;

    @Column(name = "start_date")
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    @Column(name = "registration_start_date")
    private LocalDateTime registrationStartDate;

    @Column(name = "registration_end_date")
    private LocalDateTime registrationEndDate;

    @Column(name = "max_participants")
    private Integer maxParticipants;

    private String location;

    @Column(name = "activity_type")
    private String activityType;

    @Column(name = "image_url")
    private String imageUrl;

    private String organizer;

    private Integer points;

    private String status;

    // created_by → employee.id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private Employee createdBy;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
